GameServerLib/GameObjects/AttackableUnits/AttackableUnit.cs

```diff
@@ -1487,14 +1487,22 @@ public void DashToLocation(Vector2 endPos, float dashSpeed, string animation = "
         /// </summary>
         /// <param name="state">State to set. True = dashing, false = not dashing.</param>
         /// TODO: Implement ForcedMovement methods and enumerators to handle different kinds of dashes.
-        public virtual void SetDashingState(bool state)
+        public virtual void SetDashingState(bool state, bool wasCancelled = false)
```

Comment:
I think it is better to change the state to enum. Avoid adding additional booleans as it will complicate many things in the future.

Reply:
Done i think, not sure if i did it in a right way.